client/checkout/api/index.js

```diff
@@ -0,0 +1,249 @@
+	 * @returns {Object} A request object, which will be prepared and then `.send()`.
+	 */
+	generatePaymentMethodRequest( elements, preparedCustomerData = {} ) {
+		const stripe = this.getStripe();
+
+		return new ( class {
+			constructor() {
+				this.args = {
+					type: 'card',
+					...elements,
+					// eslint-disable-next-line camelcase
+					billing_details: {
+						address: {},
+					},
+				};
+			}
+
+			/**
+			 * Prepares a value that's been loaded from inputs,
+			 * uses a default value if none is present.
+			 *
+			 * @param {string} name The key of the value.
+			 * @param {mixed} value The value to sanitize.
+			 * @returns {mixed}     The sanitized value, `undefined` if not present.
+			 */
+			prepareValue( name, value ) {
+				// Fall back to the value in `preparedCustomerData`.
+				if ( 'undefined' === typeof value || 0 === value.length ) {
+					value = preparedCustomerData[ name ]; // `undefined` if not set.
+				}
+
+				if ( 'undefined' !== typeof value && 0 < value.length ) {
+					return value;
+				}
+			}
+
+			/**
+			 * Updates a billing detail within the request.
+			 *
+			 * @param {string} name The name of the billing value.
+			 * @param {string} value The actual value.
+			 */
+			setBillingDetail( name, value ) {
+				const preparedValue = this.prepareValue( name, value );
+				if ( 'undefined' !== typeof preparedValue ) {
+					this.args.billing_details[ name ] = preparedValue;
+				}
+			}
+
+			/**
+			 * Updates an address detail within the request.
+			 *
+			 * @param {string} name The name of the address value.
+			 * @param {string} value The actual value.
+			 */
+			setAddressDetail( name, value ) {
+				const preparedValue = this.prepareValue( name, value );
+				if ( 'undefined' !== typeof preparedValue ) {
+					this.args.billing_details.address[ name ] = preparedValue;
+				}
+			}
+
+			/**
+			 * Sends the request to Stripe once everything is ready.
+			 *
+			 * @return {Object} The payment method object if successfully loaded.
+			 */
+			send() {
+				return stripe
+					.createPaymentMethod( this.args )
+					.then( ( paymentMethod ) => {
+						if ( paymentMethod.error ) {
+							throw paymentMethod.error;
+						}
+
+						return paymentMethod;
+					} );
+			}
+		} )();
+	}
+
+	/**
+	 * Extracts the details about a payment intent from the redirect URL,
+	 * and displays the intent confirmation modal (if needed).
+	 *
+	 * @param {string} redirectUrl The redirect URL, returned from the server.
+	 * @returns {mixed} A redirect URL on success, or `true` if no confirmation is needed.
+	 */
+	confirmIntent( redirectUrl ) {
+		const partials = redirectUrl.match( /#wcpay-confirm-pi:(.+):(.+):(.+)$/ );
+
+		if ( ! partials ) {
+			return true;
+		}
+
+		let orderId = partials[ 1 ];
+		const clientSecret = partials[ 2 ];
+		// Update the current order status nonce with the new one to ensure that the update
+		// order status call works when a guest user creates an account during checkout.
```

Comment:
Even after enabling create-account-during-checkout in settings, I didn't see such an option on the block based checkout. How does one enable this?

Reply:
We're working on it! https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/2377 https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2851

See also P2 – supporting signup has triggered iteration/work in a few related areas (and lots of discussion). cc @nerrad

Reply:
@allendav checking out the branch from https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2851 allows us to test this. I didn't have it in the testing instructions, because this is change that I brought over from the classic checkout and I didn't test it deeply with blocks.